packages/relay/src/lib/eth.ts

```diff
@@ -1334,10 +1335,21 @@ export class EthImpl implements Eth {
       return null;
     }
 
+    // If blocknumber requested is more than a few blocks greater than the latest, fail fast. 
+    // const latestBlock = await this.mirrorNodeClient.getLatestBlock(requestId);
+    if(!returnLatest){
+      const response = await this.mirrorNodeClient.getLatestBlock(requestId);
```

Comment:
is this returnLatest correct? I mean returnLatest = true, it wont enter this condition, and the call from getBlock is always sending true. thus never entering this condition. not sure to understand expected logic.

Reply:
Seems like `returnLatest` implies the blockTag is 'latest' in which case this flow will correctly ignore that consideration.
However where it's not i.e. `!returnLatest` it implies a blockTag had a value for a specific block to searhc for. In that case we won't to check it's within bounds and fail fast if it's not.
Is that correct @ebadiere 
If so this seems like a nice piece of logic to pull out into a method with method comments maybe `isBlockWithinBounds(blockNumberOrTag, requestId)`
Then this code become
```
if (!isBlockWithinBounds()) {
  return null;
}
```

Reply:
@ebadiere is this correct?